docs/_static/nuts-discovery.yaml

```diff
@@ -0,0 +1,115 @@
+openapi: "3.0.0"
+info:
+  title: Nuts discovery API spec
+  description: API specification for certificate related services hosted by the network authority
+  version: 0.1.0
+  license:
+    name: GPLv3
+paths:
+  /api/csr:
+    post:
+      operationId: submit
+      summary: "submit a PEM encoded CSR to the network authority"
+      tags:
+        - certificates
+      requestBody:
+        description: PEM encoded CSR
+        required: true
+        content:
+          text/plain:
+            schema:
+              type: string
+      responses:
+        '200':
+          description: "request was accepted"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/certificate_request"
+  /api/csr/{urn}:
+    get:
+      operationId: listRequests
+      summary: "list certificate signing requests"
+      tags:
+        - certificates
+      parameters:
+        - name: urn
+          in: path
+          description: "vendor oid in urn format"
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: "success"
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/certificate_request"
+  /api/x509/{urn}:
+    get:
+      operationId: listCertificates
+      summary: "get signed certificate(s) based on urn. The urn must be an oid which corresponds to the oid in the SAN"
+      tags:
+        - certificates
+      parameters:
```

Comment:
This is a REST resource for an x509 certificate, so I'd expect the identifier for the resource (path element directly after the type, 'urn') to uniquely identify an x509 certificate, instead of returning a list.

If 'vendor' identified by OID is something we consider being an entity in the Network Authority domain, I'd expect the operation to list its certificates to be something like this:

`/api/vendor/{urn}/x509`

At the same time, I'm not sure whether we should introduce the 'vendor' concept in this API. Maybe make this endpoint a sort of query endpoint?

`/api/x509/?san=some-oid`

Makes future addition easier.

Reply:
I'll do the latest: `?otherName=X`